### PR TITLE
chore(script): add option to delete logs folders on clean-up

### PR DESCRIPTION
**Pull request by serrovsky**
This PR tries to fix a problem that I had some weeks ago. If you are storing your logs locally, and since the logs folders tree structure by default follows the pattern <dag_id>/<run_id>/<task_id>/<attempt_id>.log this scales quickly. 

After some time you got thousands of empty folders and the clean-logs.sh starts to consume a lot of memory due to the find command.

image

Where you can find an example of what was happening in our case. Work-log-groomer containers were consuming 6-7 GB of memory on average just to clean logs. 🤯

With this small change, I added a flag that allows us to delete not only the file but also all the folders, once it doesn't make sense to save empty folders.

In the screenshot below you can find the real impact of this change:
image

### CODE REVIEW

File: scripts/docker/clean-logs.sh [hunk 1]
```diff
@@ -21,19 +21,30 @@ set -euo pipefail
...
+echo "Running the improved cleaning logs..."
 echo "Cleaning logs every $EVERY seconds"
 
 while true; do
-  echo "Trimming airflow logs to ${RETENTION} days."
-  find "${DIRECTORY}"/logs \
-    -type d -name 'lost+found' -prune -o \
-    -type f -mtime +"${RETENTION}" -name '*.log' -print0 | \
-    xargs -0 rm -f
+  if [ "${DELETE_LOGS_FOLDERS}" = true ]
```
Comment (potiuk) [inline scripts/docker/clean-logs.sh]:
```suggestion
  if [[ "${DELETE_LOGS_FOLDERS}" == "true" ]]
```

Comment (potiuk) [reply scripts/docker/clean-logs.sh]: NIT: We are trying to follow "Google bash standards for bash scripts

**Review by potiuk — COMMENTED** [review]
Nice improvement.

**Review by potiuk — APPROVED** [review]
Small nit but otherwise cool

File: scripts/docker/clean-logs.sh (hunk 1 above)
Comment (serrovsky) [reply scripts/docker/clean-logs.sh]: Thanks for the review. Done

File: scripts/docker/clean-logs.sh [hunk 2]
```diff
@@ -21,19 +21,30 @@ set -euo pipefail
 
 readonly DIRECTORY="${AIRFLOW_HOME:-/usr/local/airflow}"
 readonly RETENTION="${AIRFLOW__LOG_RETENTION_DAYS:-15}"
+readonly DELETE_LOGS_FOLDERS="${AIRFLOW__LOG_DELETE_FOLDERS:-true}"
 
 trap "exit" INT TERM
 
 readonly EVERY=$((15*60))
 
+echo "Running the improved cleaning logs..."
```
Comment (jedcunningham) [inline scripts/docker/clean-logs.sh:30]:
```suggestion
```

Leftover from development? I don't think we need this :)

File: scripts/docker/clean-logs.sh [hunk 3]
```diff
@@ -21,19 +21,30 @@ set -euo pipefail
...
-  find "${DIRECTORY}"/logs \
-    -type d -name 'lost+found' -prune -o \
-    -type f -mtime +"${RETENTION}" -name '*.log' -print0 | \
-    xargs -0 rm -f
+  if [[ "${DELETE_LOGS_FOLDERS}" == "true" ]]
+  then
+    echo "Trimming airflow logs folders to ${RETENTION} days."
+    find "${DIRECTORY}"/logs \
+      -type d -name 'lost+found' -prune -o \
+      -type d -mtime +"${RETENTION}" -print0 | \
```
Comment (jedcunningham) [inline scripts/docker/clean-logs.sh:39]: This won't work as written. mtime updates don't propogate up, so you'd end up deleting all of a dags logs after the root folder has been around for $retention_days.

Instead, we could have a second command that goes through and deletes old empty folders? I think the "empty" aspect is critical.

File: scripts/docker/clean-logs.sh [hunk 4]
```diff
@@ -21,19 +21,30 @@ set -euo pipefail
 
 readonly DIRECTORY="${AIRFLOW_HOME:-/usr/local/airflow}"
 readonly RETENTION="${AIRFLOW__LOG_RETENTION_DAYS:-15}"
+readonly DELETE_LOGS_FOLDERS="${AIRFLOW__LOG_DELETE_FOLDERS:-true}"
```
Comment (jedcunningham) [inline scripts/docker/clean-logs.sh:24]: I almost feel like this should just happen regardless and not be able to be turned off. @potiuk, thoughts?

File: chart/values.yaml [hunk 5]
```diff
@@ -650,6 +650,7 @@ workers:
     args: ["bash", "/clean-logs"]
     # Number of days to retain logs
     retentionDays: 15
+    deleteLogFolders: true
```
Comment (jedcunningham) [inline chart/values.yaml:653]: If we do keep it confugurable, I think `deleteFolders` is better. Like with `retentionDays`, we know we are dealing with logs already.

**Review by jedcunningham — CHANGES_REQUESTED** [review]

File: scripts/docker/clean-logs.sh (hunk 3 above)
Comment (potiuk) [reply scripts/docker/clean-logs.sh:39]: Good point @jedcunningham  adding `-empty` predicate in find should solve the issue.

Comment (potiuk) [inline scripts/docker/clean-logs.sh:39]:
```suggestion
      -type d -empty -mtime +"${RETENTION}" -print0 | \
```

**Pull request closed without merging**